src/Elm/Print.hs

```diff
@@ -528,6 +550,14 @@ typeDecoderDoc  t@ElmType{..} =
     enumDecoder :: Doc ann
     enumDecoder = name <+> "elmStreetDecodeEnum read" <> typeName
 
+    newtypeDecoder :: Doc ann
+    newtypeDecoder = name <+> "D.map" <+> typeName <+> fieldDecoderDoc
+      where
+        fieldDecoderDoc :: Doc ann
+        fieldDecoderDoc = case elmConstructorFields $ NE.head elmTypeConstructors of
+            []    -> "ERROR"
```

Comment:
Do you think it's better to write it as the Decoder error: `D.fail "Bla bla error"`?

Reply:
Good idea 👍 With `Decoder` we actually have opportunity to report errors. Can can